spec/lib/content_security_policy_spec.rb

```diff
@@ -303,13 +303,23 @@ def theme_policy
       theme.theme_modifier_set.csp_extensions = ["script-src: https://from-theme-flag.script", "worker-src: from-theme-flag.worker"]
       theme.save!
 
+      child_theme = Fabricate(:theme, component: true)
+      theme.add_relative_theme!(:child, child_theme)
+      child_theme.theme_modifier_set.csp_extensions = ["script-src: https://child-theme-flag.script", "worker-src: child-theme-flag.worker"]
+      child_theme.save!
+
       expect(parse(theme_policy)['script-src']).to include('https://from-theme-flag.script')
+      expect(parse(theme_policy)['script-src']).to include('https://child-theme-flag.script')
       expect(parse(theme_policy)['worker-src']).to include('from-theme-flag.worker')
+      expect(parse(theme_policy)['worker-src']).to include('child-theme-flag.worker')
```

Comment:
@davidtaylorhq I've improved our test suite here which didn't catch csp_extensions that are set in a child theme.